pkg/metrics/metrics.go

```diff
@@ -175,3 +305,25 @@ func (opMgr *operationMetricsManager) StartMetricsEndpoint(pattern, addr string,
 	}()
 	return srv, nil
 }
+
+// snapshotProvisionType represents which kind of snapshot a metric is
+type snapshotProvisionType string
+
+// snapshotStatusType represents the type of snapshot status to report
+type snapshotStatusType string
+
+// SnapshotOperationStatus represents the status for a snapshot controller operation
+type SnapshotOperationStatus struct {
+	statusCode snapshotStatusType
```

Comment:
I think the string simplifies things when it comes to recording the metrics. Otherwise, we would need to have a switch statement to get the string version of an enum. I can switch to that model if needed though.